packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,395 @@
+        metadata: this.options?.metadata?.map(metadata => metadata.bind(scope).metadataMatch),
+      },
+    };
+  }
+}
+
+/**
+ * All properties for gRPC gateway route rewrite.
+ */
+export interface HttpGatewayRouteRewriteConfig {
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route host name rewrite.
+   *
+   * @default - rewrite hostname to the destination Virtual Service name.
+   */
+  readonly hostname?: CfnGatewayRoute.GatewayRouteHostnameRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route path rewrite.
+   *
+   * @default - rewrite path to '/'.
+   */
+  readonly path?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefix?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines gRPC gateway route rewrite that can be defined in addition to host name.
+ */
+export abstract class HttpGatewayRoutePathOrPrefixRewrite {
+  /**
+   * The path to rewrite.
+   *
+   * @param exact The exact path to rewrite.
+   */
+  public static path(exact: string): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePathRewriteImpl(exact);
+  }
+
+  /**
+   * The default prefix used to replace the incoming route prefix when rewritten.
+   *
+   * @param isDefault Boolean to select to enable or disable the default prefix. Default value is enabled.
+   */
+  public static defaultPrefix(isDefault?: boolean): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ defaultPrefix: isDefault == false ? 'DISABLED' : 'ENABLED' });
+  }
+
+  /**
+   * Replace the incoming route prefix when rewritten.
+   *
+   * @param value The value used to replace the incoming route prefix when rewritten.
+   */
+  public static customPrefix(value: string): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ value: value } );
+  }
+
+  /**
+   * Return HTTP gateway route rewrite configuration.
    */
-  readonly prefixPath: string;
+  abstract bind(scope: Construct): HttpGatewayRouteRewriteConfig;
+}
+
+class HttpGatewayRoutePathRewriteImpl extends HttpGatewayRoutePathOrPrefixRewrite {
+  constructor(
+    private readonly exact: string,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRouteRewriteConfig {
+    return {
+      path: {
+        exact: this.exact,
+      },
+    };
+  }
+}
+
+class HttpGatewayRoutePrefixRewriteImpl extends HttpGatewayRoutePathOrPrefixRewrite {
+  constructor(
+    private readonly prefixRewrite: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRouteRewriteConfig {
+    return {
+      prefix: this.prefixRewrite,
+    };
+  }
+}
+
+/**
+ * The criterion for determining a rewrite for this HTTP GatewayRoute.
+ */
+export interface HttpGatewayRouteRewrite {
```

Comment:
I don't like this interface. Seems redundant to me.

Remove it, and add these two new properties directly to `HttpGatewayRouteSpecOptions` (with possibly slightly different names, maybe including the word "rewrite" somewhere).

Reply:
Well noted.